mobile/native/src/dlc_handler.rs

```diff
@@ -0,0 +1,158 @@
+                if let Err(e) = dlc_handler.send_dlc_message(peer, msg) {
+                    tracing::error!(peer=%peer, "Failed to process end dlc message event. {e:#}");
+                }
+            }
+            Err(RecvError::Lagged(skipped)) => {
+                tracing::warn!("Skipped {skipped} messages");
+            }
+            Err(RecvError::Closed) => {
+                tracing::error!("Lost connection to sender!");
+                break;
+            }
+        }
+    }
+}
+
+impl DlcHandler {
+    pub fn send_dlc_message(&self, peer: PublicKey, msg: Message) -> Result<()> {
+        let mut conn = db::connection()?;
+
+        let serialized_outbound_message = SerializedDlcMessage::try_from(&msg)?;
+        let outbound_msg = DlcMessage::new(peer, serialized_outbound_message.clone(), false)?;
+
+        db::dlc_messages::DlcMessage::insert(&mut conn, outbound_msg)?;
+        db::last_outbound_dlc_messages::LastOutboundDlcMessage::upsert(
+            &mut conn,
+            &peer,
+            serialized_outbound_message,
+        )?;
+
+        send_dlc_message(
+            &self.node.dlc_message_handler,
+            &self.node.peer_manager,
+            peer,
+            msg,
+        );
+
+        Ok(())
+    }
+
+    pub fn on_connect(&self, peer: PublicKey) -> Result<()> {
+        let mut conn = db::connection()?;
+        let last_outbound_serialized_dlc_message =
+            db::last_outbound_dlc_messages::LastOutboundDlcMessage::get(&mut conn, &peer)?;
+
+        if let Some(last_outbound_serialized_dlc_message) = last_outbound_serialized_dlc_message {
+            tracing::debug!(%peer, ?last_outbound_serialized_dlc_message.message_type, "Sending last dlc message");
+
+            let message = Message::try_from(&last_outbound_serialized_dlc_message)?;
+            send_dlc_message(
+                &self.node.dlc_message_handler,
+                &self.node.peer_manager,
+                peer,
+                message,
+            );
+        } else {
+            tracing::debug!(%peer, "No last dlc message found. Nothing todo.");
+        }
+
+        Ok(())
+    }
+    // Returns either the dlc message step or return none, if the dlc message has already been
+    // processed.
+    pub fn start_dlc_message_step(
+        conn: &mut SqliteConnection,
+        msg: &Message,
+        peer_id: PublicKey,
+    ) -> Result<Option<DlcMessageStep>> {
+        let serialized_inbound_message = SerializedDlcMessage::try_from(msg)?;
+        let inbound_msg = DlcMessage::new(peer_id, serialized_inbound_message, true)?;
+
+        let dlc_message_step =
+            match db::dlc_messages::DlcMessage::get(conn, inbound_msg.message_hash)? {
+                Some(_) => None, // the dlc message has already been processed, no step necessary.
+                None => Some(DlcMessageStep {
+                    inbound_msg,
+                    peer_id,
+                }),
```

Comment:
Is the reason for having this code duplicated this part where we store things in the DB? I.e. because we have different databases for coordinator and the app?

Reply:
I guess my comment above on the other `pub fn start_dlc_message_step(` would help removing some duplications.

Reply:
> Is the reason for having this code duplicated this part where we store things in the DB? I.e. because we have different databases for coordinator and the app?

yes, that's the reason why the `dlc_handler.rs` looks almost identical on the app and the coordinator.

> I guess my comment above on the other pub fn start_dlc_message_step( would help removing some duplications.

Thanks for the suggestion, I've removed the `dlc_message_step` part, which indeed removed some duplicate code :).